parts/k8s/kubernetesagentresourcesvmas.t

```diff
@@ -5,20 +5,17 @@
         "name": "loop"
       },
       "dependsOn": [
-{{if .IsCustomVNET}}
-      "[variables('nsgID')]"
-{{else}}
-      "[variables('vnetID')]"
+{{if not .IsCustomVNET}}
+      "[variables('vnetID')]",
 {{end}}
+      "[variables('nsgID')]"
       ],
       "location": "[variables('location')]",
       "name": "[concat(variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
       "properties": {
-{{if .IsCustomVNET}}
```

Comment:
Before: if custom VNET attach a NSG to the agent NIC

After: attach a NSG to the agent NIC _always_